### PR TITLE
Refactor forecast output

### LINKED ISSUE #199: Change forecast result to just distributions?
I noticed that the forecast function returns `mean.(d), d`. What do you think of just returning the distribution `d`? and leave the statistic of interest to the user? I think it would be more elegant, and would make users think about what they are getting from the function.

### PR DESCRIPTION
**Pull request by juliohm**
Fix #199 

Thanks for considering the change. Besides refactoring the output, I also took the opportunity to refactor the way the distribution vector was being constructed. The new way ensures that the eltype is a MvNormal as opposed to a generic Distribution.

Before I made the changes one test was already failing locally for me. It should be unrelated to the PR.

### CODE REVIEW

File: src/forecast.jl [hunk 1]
```diff
@@ -3,7 +3,7 @@ export forecast, simulate
 """
     forecast(ss::StateSpace{Typ}, N::Int) where Typ
 
-Obtain the minimum mean square error forecasts N steps ahead. Returns the forecasts and the predictive distributions 
+Obtain the minimum mean square error forecasts N steps ahead. Returns the forecasts and the predictive distributions
```
Comment (raphaelsaavedra) [inline src/forecast.jl:6]:
```suggestion
Obtain the minimum mean square error forecasts `N` steps ahead, returning the predictive distributions.
```

**Review by raphaelsaavedra — COMMENTED** [review]
Thanks for taking the time :) please also bump the package version and I'm happy to merge this and tag a new release.

**Comment by raphaelsaavedra** [comment]
> Before I made the changes one test was already failing locally for me

That's interesting, I wonder if your project wasn't set up exactly as the package's? Tests are passing locally for me and also passing in CI

**Comment by juliohm** [comment]
> That's interesting, I wonder if your project wasn't set up exactly as the package's? Tests are passing locally for me and also passing in CI

I cloned the repo, and instantiated the environment of the package with:

```Julia
$ cd /path/to/cloned/repo
$ julia --project
] instantiate
] test
```

**Comment by juliohm** [comment]
Bumped the version as suggested 👌

**Comment by raphaelsaavedra** [comment]
I think I know what it is. You're probably using Julia 1.5 which means https://github.com/LAMPSPUC/StateSpaceModels.jl/issues/155 is probably causing a test to fail (I'm not sure why this issue was closed, I don't think we addressed this). It's caused by the recent change in RNG in `Random`.

**Comment by juliohm** [comment]
I can confirm I am on Julia v1.5 👍

**Pull request merged**